src/History.js

```diff
@@ -175,17 +175,17 @@ var EnyoHistory = module.exports = kind.singleton(
 	},
 
 	/**
-	* Asynchronously drops all history entries without calling their respective handlers. When the
+	* Clears all history entries without calling their respective handlers. When the
 	* entries are popped, the internal history will be empty and the browser history will be
 	* reset to the state when this module started tracking the history.
 	*
 	* @public
 	*/
 	clear: function () {
-		var len = _history.length;
-		if (len) {
-			this.drop(len);
-		}
+		_queue.length = 0;
```

Comment:
I think we need to run through and pop off the array to ensure that we aren't inadvertently keeping orphaned objects in memory. If the array were merely simple types, this would be an ok performance hack.